client/core/core.go

```diff
@@ -3630,13 +3630,19 @@ func (c *Core) resumeTrades(dc *dexConnection, trackers []*trackedTrade) assetMa
 					notifyErr(SubjectMatchStatusError, "Match %s for order %s is in state %s, but has no maker swap coin.", dbMatch.Side, tracker.token(), dbMatch.Status)
 					continue
 				}
-				counterContract := metaData.Proof.CounterScript
+				counterContract := metaData.Proof.CounterContract
 				if len(counterContract) == 0 {
 					match.swapErr = fmt.Errorf("missing counter-contract, order %s, match %s", tracker.ID(), match.id)
 					notifyErr(SubjectMatchStatusError, "Match %s for order %s is in state %s, but has no maker swap contract.", dbMatch.Side, tracker.token(), dbMatch.Status)
 					continue
 				}
-				auditInfo, err := wallets.toWallet.AuditContract(counterSwap, counterContract)
+				counterTxData := metaData.Proof.CounterTxData
+				if len(counterTxData) == 0 {
+					match.swapErr = fmt.Errorf("missing counter-tx data, order %s, match %s", tracker.ID(), match.id)
+					notifyErr("Match status error", "Match %s for order %s is in state %s, but has no maker swap tx data.", dbMatch.Side, tracker.token(), dbMatch.Status)
+					continue
```

Comment:
I got the impression from the other changes that this PR would make the client able to work with a server that does not yet provide the `CounterTxData`.  https://github.com/decred/dcrdex/pull/804/files#diff-8962b4c526755f0f98ed9f46ae05827e8edbc43ed5671ea4a2624bdf1617496cR2029 (`(*trackedTrade).auditContract`).  Should this be an error and continue to next match, as you have it, or just pass nil to `AuditContract` and try to work with it?  Ofc it will `AuditContract` doesn't use it yet so it will be fine, but I figure this will be the upgrade plan where the server is behind clients.

Reply:
Yeah. I got a little overzealous here. This is going to be tricky without versioning, because we can't allow the user of SPV wallets on pre-tx-data servers.